src/reynir_correct/spelling.py

```diff
@@ -635,20 +659,23 @@ def stupid_backoff(w: str) -> float:
                         i + 1, word, c, log_prob
                     )
                 )
-        # Find the candidate with the highest probability
-        m = max(candidates, key=lambda t: t[1])
-        if m[1] < self._MIN_LOG_PROBABILITY and (
-            word in self.ngrams or original_word in self.ngrams
-        ):
-            # Best candidate is very unlikely: return the original word
-            # print(f"Best candidate {m[0]} is highly unlikely, returning original {word}")
-            return word
-        # Return the most likely word
-        return m[0]
+        # Skip very unlikely candidates
+        best_candidates = list(
+            [
+                x
+                for x in candidates
+                if not (
+                    x[1] < self._MIN_LOG_PROBABILITY
+                    and (word in self.ngrams or original_word in self.ngrams)
```

Comment:
This is incorrect. It should in any case be `x` and not `word`, but also the functionality of the original code was different: if the correction candidate is very unlikely and the original word exists as a unigram in the ngrams database, the original word is returned instead of the correction candidate, i.e. no correction or suggestion is made.

Reply:
Changed behaviour; now the functions for returning a list and a single candidate have been split up in a more definitive way. For the list, if the top candidate fulfills this if statement, the assumption is that the other candidates also fail, so it returns an empty list.